File: ios/Runner/ExerciseCoaching/BaseExerciseAnalyzer.swift
import Foundation
import MLKitPoseDetection

/// Egzersiz analiz sonucunu temsil eden yapı.
struct AnalysisResult {
    let accuracy: Double      // 0.0 - 1.0
    let feedback: String      // Kullanıcıya gösterilecek mesaj
    let isRepetitionComplete: Bool // Tekrar tamamlandı mı?
    let debugInfo: [String: Any]?  // Debug overlay için ek bilgi
    
    init(accuracy: Double, feedback: String, isRepetitionComplete: Bool = false, debugInfo: [String: Any]? = nil) {
        self.accuracy = accuracy
        self.feedback = feedback
        self.isRepetitionComplete = isRepetitionComplete
        self.debugInfo = debugInfo
    }
}

/// Tüm egzersiz analizörlerinin protokolü.
/// Android'deki BaseExerciseAnalyzer.java karşılığı.
protocol ExerciseAnalyzer: AnyObject {
    /// Poz verisini analiz eder ve sonuç döndürür
    func analyze(pose: Pose) -> AnalysisResult
    
    /// Analizörü sıfırlar (yeni egzersiz başladığında)
    func reset()
}

/// Varsayılan reset implementasyonu
extension ExerciseAnalyzer {
    func reset() {
        // Override edilebilir
    }
}

File: android/app/src/main/java/com/breathflow/app/java/exercise_coaching/exercises/BaseExerciseAnalyzer.java
package com.breathflow.app.java.exercise_coaching.exercises;

import com.google.mlkit.vision.pose.Pose;

/**
 * Tüm egzersiz analizörlerinin temel sınıfı.
 * Kullanıcı kendi egzersiz dosyalarını yazarak bu sınıfı extend edecek.
 */
public abstract class BaseExerciseAnalyzer {

    /**
     * Analiz sonucunu temsil eden sınıf
     */
    public static class AnalysisResult {
        public final double accuracy; // 0.0 - 1.0
        public final String feedback; // Kullanıcıya gösterilecek mesaj
        public final boolean isRepetitionComplete; // Tekrar tamamlandı mı?

        public AnalysisResult(double accuracy, String feedback, boolean isRepetitionComplete) {
            this.accuracy = accuracy;
            this.feedback = feedback;
            this.isRepetitionComplete = isRepetitionComplete;
        }

        public AnalysisResult(double accuracy, String feedback) {
            this(accuracy, feedback, false);
        }
    }

    /**
     * Poz verisini analiz eder ve sonuç döndürür
     *
     * @param pose MLKit Pose nesnesi
     * @return Analiz sonucu (accuracy, feedback, repetition)
     */
    public abstract AnalysisResult analyze(Pose pose);

    /**
     * Analizörü sıfırlar (yeni egzersiz başladığında)
     */
    public void reset() {
        // Override edilebilir
    }
}
